logwatch/logwatch.go

```diff
@@ -172,22 +212,117 @@ func (m *LogWatch) ConnectContainer(ctx context.Context, container testcontainer
 	m.log.Info().
 		Str("Prefix", prefix).
 		Str("Name", name).
+		Str("Timeout", m.logProducerTimeout.String()).
 		Msg("Connecting container logs")
 	m.consumers[name] = cons
 	m.containers = append(m.containers, container)
 	container.FollowOutput(cons)
-	return container.StartLogProducer(ctx)
+	err = container.StartLogProducer(ctx, m.logProducerTimeout)
+
+	go func(done chan struct{}, timeout time.Duration, retryLimit int) {
+		defer m.log.Info().Str("Container name", name).Msg("Disconnected container logs")
+		currentAttempt := 1
+
+		var shouldRetry = func() bool {
+			if retryLimit == -1 {
+				return true
+			}
+
+			if currentAttempt < retryLimit {
+				currentAttempt++
+				return true
+			}
+
+			return false
+		}
+
+		for {
+			select {
+			case logErr := <-container.GetLogProducerErrorChannel():
+				if logErr != nil {
+					m.log.Error().
+						Err(err).
+						Str("Container name", name).
+						Msg("Log producer errored")
+					if shouldRetry() {
```

Comment:
I didn't know it existed 😅

Reply:
I did use it in one place, but didn't replace all usages, because in some I want to pass incremental timeout, not just execute with the same value again and `avast/retry` doesn't support such a thing